core/App/screens/PinEnter.tsx

```diff
@@ -26,15 +34,70 @@ const PinEnter: React.FC<PinEnterProps> = ({ setAuthenticated }) => {
     },
   })
 
-  const checkPin = async (pin: string) => {
-    const keychainEntry = await Keychain.getGenericPassword({ service: 'passcode' })
-    if (keychainEntry && JSON.stringify(pin) === keychainEntry.password) {
-      setAuthenticated(true)
+  const onPinChanged = async (pin: string) => {
+    if (authLevel === AuthLevel.BiometricsAndPin) {
+      if (walletSecret) {
+        const generatedKey = await getKeyForPIN(pin)
+        if (generatedKey === walletSecret.walletKey) {
+          setAuthenticated(true)
+        } else {
+          setModalVisible(true)
+        }
+      } else {
+        // TODO: Error handling
+        Alert.alert('Error: Wallet secret undefined!.\nPlease reload the app')
+      }
+      await generateKeyForPIN(pin)
     } else {
-      setModalVisible(true)
+      // Fallback to PIN, attempt to init wallet with generated key
+      const isError = await checkPIN(pin)
+
+      if (isError) {
+        setModalVisible(true)
+        return
+      } else {
+        setAuthenticated(true)
+      }
+    }
+  }
+
+  const initWithBiometrics = async () => {
+    // TODO: get auth level from settings
+    // const authLevel = await getAuthLevel()
```

Comment:
Will this be required in the future?

Reply:
Yes, it is hardcoded now but next level it should be configured in the settings or at least an environment variable